c/macOS/src/main.c

```diff
@@ -0,0 +1,90 @@
+//
+//  main.c
+//  mptcp-network-client
+//
+//  Created by Anthony Doeraene on 08/08/2024.
+//
+
+#include <stdio.h>
+#include <string.h>
+#include <Network/Network.h>
+
+dispatch_semaphore_t semaphore;
+
+void get(char *path, nw_connection_t conn, dispatch_queue_t queue){
+    char req[1024];
+    int size = sprintf(req, "GET %s HTTP/1.0\r\nUser-Agent: curl/1.0.0\r\n\n", path);
+    
+    dispatch_data_t data = dispatch_data_create(req, size, queue, ^{
+    });
+    printf("Sending request...\n");
+    nw_content_context_t context = nw_content_context_create("context");
+    nw_connection_send(conn, data, context, true, ^(nw_error_t  _Nullable error) {
```

Comment:
This `^(...) { ... }` is an interesting technique, but it doesn't look like C :)

Reply:
Seems like it is an extension done to C by apple, but I don't really know a lot more about it. Perhaps putting it in the objective-c folder instead ?

Reply:
Quite funny however that gcc still manage to compile this with the c extension :D

Reply:
Interesting, I would not be surprised if it is specific to them!